build-tools/packages/build-cli/src/commands/release/setDtsTypes.ts

```diff
@@ -0,0 +1,210 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags } from "@oclif/core";
+import { Package, updatePackageJsonFile } from "@fluidframework/build-tools";
+import { PackageCommand } from "../../BasePackageCommand";
+// eslint-disable-next-line node/no-missing-import
+import type { PackageJson } from "type-fest";
+import { ExtractorConfig } from "@microsoft/api-extractor";
+import { CommandLogger } from "../../logging";
+import path from "node:path";
+
+/**
+ * Represents a list of package categorized into two arrays
+ */
+interface PackageTypesList {
+	/**
+	 * An array of strings containing package names that have been updated.
+	 */
+	packagesUpdated: string[];
+	/**
+	 * An array of strings containing package names that have not been updated.
+	 */
+	packagesNotUpdated: string[];
+}
+
+/**
+ * Represents a configuration object for updating a package.
+ */
+interface ReleaseTagConfig {
+	/**
+	 * The package to be updated.
+	 */
+	pkg: Package;
+	/**
+	 * ExtractorConfig represents api-extractor.json config file.
+	 */
+	extractorConfig: ExtractorConfig;
+	/**
+	 * The type of release for the package (e.g., "alpha", "beta", "public", "default").
+	 */
+	releaseType: ReleaseTag;
+	/**
+	 * The JSON data for the package.
+	 */
+	json: PackageJson;
+}
```

Comment:
This interface doesn't seem necessary. Since it's only used in one function, I would just make these all parameters that are passed into the function directly rather than wrapping them.